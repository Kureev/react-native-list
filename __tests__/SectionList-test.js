/**
 * Copyright (c) 2015-present, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree. An additional grant
 * of patent rights can be found in the PATENTS file in the same directory.
 *
 */
'use strict';

jest.disableAutomock();

const React = require('React');
const ReactTestRenderer = require('react-test-renderer');

const SectionList = require('SectionList');

describe('SectionList', () => {
  it('renders empty list', () => {
    const component = ReactTestRenderer.create(
      <SectionList
        sections={[]}
        renderItem={({item}) => <item value={item.key} />}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('rendering empty section headers is fine', () => {
    const component = ReactTestRenderer.create(
      <SectionList
        sections={[{key: 's1', data: [{key: 'i1'}, {key: 'i2'}]}]}
        renderItem={({item}) => <item value={item.key} />}
        renderSectionHeader={() => null}
      />
    );
    expect(component).toMatchSnapshot();
  });
  it('renders all the bells and whistles', () => {
    const component = ReactTestRenderer.create(
      <SectionList
        ItemSeparatorComponent={() => <defaultItemSeparator />}
        ListFooterComponent={() => <footer />}
        ListHeaderComponent={() => <header />}
        SectionSeparatorComponent={() => <sectionSeparator />}
        sections={[
          {
            renderItem: ({item}) => <itemForSection1 value={item.id} />,
            key: '1st Section',
            keyExtractor: (item, index) => item.id,
            ItemSeparatorComponent: () => <itemSeparatorForSection1 />,
            data: [{id: 'i1s1'}, {id: 'i2s1'}],
          },
          {
            key: '2nd Section',
            data: [{key: 'i1s2'}, {key: 'i2s2'}],
          },
        ]}
        refreshing={false}
        onRefresh={jest.fn()}
        renderItem={({item}) => <defaultItem value={item.key} />}
        renderSectionHeader={({section}) => <sectionHeader value={section.key} />}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
