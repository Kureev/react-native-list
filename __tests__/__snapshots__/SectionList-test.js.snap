// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionList rendering empty section headers is fine 1`] = `
<RCTScrollView
  ItemSeparatorComponent={undefined}
  data={
    Array [
      Object {
        "data": Array [
          Object {
            "key": "i1",
          },
          Object {
            "key": "i2",
          },
        ],
        "key": "s1",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  renderSectionHeader={[Function]}
  scrollEventThrottle={50}
  sections={
    Array [
      Object {
        "data": Array [
          Object {
            "key": "i1",
          },
          Object {
            "key": "i2",
          },
        ],
        "key": "s1",
      },
    ]
  }
  stickyHeaderIndices={
    Array [
      0,
    ]
  }
  stickySectionHeadersEnabled={true}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
    />
    <View
      onLayout={[Function]}
    >
      <item
        value="i1"
      />
    </View>
    <View
      onLayout={[Function]}
    >
      <item
        value="i2"
      />
    </View>
  </View>
</RCTScrollView>
`;

exports[`SectionList renders all the bells and whistles 1`] = `
<RCTScrollView
  ItemSeparatorComponent={undefined}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  SectionSeparatorComponent={[Function]}
  data={
    Array [
      Object {
        "ItemSeparatorComponent": [Function],
        "data": Array [
          Object {
            "id": "i1s1",
          },
          Object {
            "id": "i2s1",
          },
        ],
        "key": "1st Section",
        "keyExtractor": [Function],
        "renderItem": [Function],
      },
      Object {
        "data": Array [
          Object {
            "key": "i1s2",
          },
          Object {
            "key": "i2s2",
          },
        ],
        "key": "2nd Section",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onRefresh={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onViewableItemsChanged={undefined}
  refreshControl={
    <RefreshControlMock
      onRefresh={[Function]}
      refreshing={false}
    />
  }
  refreshing={false}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  renderSectionHeader={[Function]}
  scrollEventThrottle={50}
  sections={
    Array [
      Object {
        "ItemSeparatorComponent": [Function],
        "data": Array [
          Object {
            "id": "i1s1",
          },
          Object {
            "id": "i2s1",
          },
        ],
        "key": "1st Section",
        "keyExtractor": [Function],
        "renderItem": [Function],
      },
      Object {
        "data": Array [
          Object {
            "key": "i1s2",
          },
          Object {
            "key": "i2s2",
          },
        ],
        "key": "2nd Section",
      },
    ]
  }
  stickyHeaderIndices={
    Array [
      1,
      4,
    ]
  }
  stickySectionHeadersEnabled={true}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <RCTRefreshControl />
  <View>
    <View
      onLayout={[Function]}
    >
      <header />
    </View>
    <View
      onLayout={[Function]}
    >
      <sectionHeader
        value="1st Section"
      />
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <sectionSeparator />
        <itemForSection1
          value="i1s1"
        />
        <itemSeparatorForSection1 />
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <itemForSection1
          value="i2s1"
        />
        <sectionSeparator />
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <sectionHeader
        value="2nd Section"
      />
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <sectionSeparator />
        <defaultItem
          value="i1s2"
        />
        <defaultItemSeparator />
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <defaultItem
          value="i2s2"
        />
        <sectionSeparator />
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <footer />
    </View>
  </View>
</RCTScrollView>
`;

exports[`SectionList renders empty list 1`] = `
<RCTScrollView
  ItemSeparatorComponent={undefined}
  data={Array []}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onViewableItemsChanged={undefined}
  renderItem={[Function]}
  renderScrollComponent={[Function]}
  scrollEventThrottle={50}
  sections={Array []}
  stickyHeaderIndices={Array []}
  stickySectionHeadersEnabled={true}
  updateCellsBatchingPeriod={50}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
